### PR TITLE
[Arm64] JIT generate LSE Atomics

### PR DESCRIPTION
**Pull request by sdmaclea**
When ARMv8.1 LSE Atomics are available, use them to implement
Interlocked operations.

@briansull @BruceForstall @CarolEidt @dotnet/jit-contrib

### CODE REVIEW

**Comment by sdmaclea** [comment]
All 22 checks passed.  Merge conflict with #18267 forced a rebase.

ping @briansull @BruceForstall @CarolEidt @dotnet/jit-contrib

File: src/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -2664,92 +2664,118 @@ void CodeGen::genLockedInstructions(GenTreeOp* treeNode)
...
 
-    noway_assert(addrReg != storeDataReg);
-    noway_assert((treeNode->OperGet() == GT_XCHG) || (addrReg != dataReg));
+        switch (treeNode->gtOper)
+        {
+            case GT_XCHG:
+                getEmitter()->emitIns_R_R_R(INS_swpal, dataSize, dataReg, targetReg, addrReg);
+                break;
+            case GT_XADD:
+                if ((targetReg == REG_NA) || (targetReg == REG_ZR))
```
Comment (mikedn) [inline src/jit/codegenarm64.cpp]: Can `targetReg` really be `REG_NA`?!

Comment (mikedn) [reply src/jit/codegenarm64.cpp]: Looks like `staddl` is the arm64 equivalent of x64's `lock add`. If you want to support this you need to generate `GT_LOCKADD` in lowering like it is done on x64. See #18267

Comment (sdmaclea) [reply src/jit/codegenarm64.cpp]: If the result is not used, it certainly could be REG_NA/REG_ZR.  I doubt lsra and/or lowering will generate this case yet.

Comment (sdmaclea) [reply src/jit/codegenarm64.cpp]: Before the rebase this used to include GT_LOCKADD.   I will look at #18267

**Review by briansull — APPROVED** [review]

**Comment by BruceForstall** [comment]
@briansull you approved this; can it be merged?

**Pull request merged**